index.js

```diff
@@ -36,17 +38,30 @@ async function run() {
       pull_number: github.context.payload.pull_request.number,
     }
 
-    const titlePrefix = inputs.titleTemplate.replace(tokenRegex, match(inputs.uppercaseTitle));
+    const title = github.context.payload.pull_request.title;
+
+    const setTitle = inputs.titleReplaceTemplate ? inputs.titleReplaceTemplate.replace(tokenRegex, match(inputs.uppercaseTitle)) : null;
+    core.debug(`setTitle: ${setTitle}`);
+
+    const setTitle = setTitle && title !== setTitle
+    if (setTitle) {
+      request.title = setTitle;
+      core.debug(`new title: ${request.title}`);
+    } else {
+      core.warning('PR title set not requested or set already - no updates made');
+    }
+
+    const configuredTitleTemplate = inputs.titlePrefixTemplate || inputs.titleTemplate;
+    const titlePrefix = configuredTitleTemplate ? configuredTitleTemplate.replace(tokenRegex, match(inputs.uppercaseTitle)) : null;
     core.debug(`titlePrefix: ${titlePrefix}`);
 
-    const title = github.context.payload.pull_request.title;
-    const updateTitle = !title.toLowerCase().startsWith(titlePrefix.toLowerCase());
+    const prefixTitle = titlePrefix && !(request.title || title).toLowerCase().startsWith(titlePrefix.toLowerCase());
```

Comment:
Actually, thinking about this, enabling *both* prefixing *and/or* replacing the title would not only complicate reasoning about the code here but also testing it. I can't think of a scenario where you'd need to specify templates for doing both. Wouldn't replacing the title make prefixing redundant?